### PR TITLE
[Plugin] Support stable diffusion.cpp

### PR DESCRIPTION
**Pull request by grorge123**

### CODE REVIEW

**Comment by juntao** [comment]
Hello, I am a [code review bot](https://github.com/flows-network/github-pr-summary/) on [flows.network](https://flows.network/).

It could take a few minutes for me to analyze this PR. Relax, grab a cup of coffee and check back later. Thanks!

**Review by hydai — CHANGES_REQUESTED** [review]

**Review by hydai — CHANGES_REQUESTED** [review]

**Comment by hydai** [comment]
Hi @grorge123 
Please fix the linter issue and rebase to the latest master, thanks.

**Comment by hydai** [comment]
@q82419 
Will the test verify the number of the mock plugins?

File: test/plugins/wasmedge_stablediffusion/wasmedge_stablediffusion.cpp [hunk 1]
```diff
@@ -0,0 +1,322 @@
...
+            31,
+            1,
+            0,
+            0,
+            0,
+            0,
+            SessionPtr}, // NThreads=-1,
+                         //  wtype=31(SD_TYPE_COUNT), RngType=CUDA_RNG,
+                         //  Schedule=DEFAULT, Other is false
+        Errno));
```
Comment (dannypsnl) [inline test/plugins/wasmedge_stablediffusion/wasmedge_stablediffusion.cpp:259]: I think a better approach is comment beside each argument, so ppl no need to compare.

File: test/plugins/wasmedge_stablediffusion/wasmedge_stablediffusion.cpp [hunk 2]
```diff
@@ -0,0 +1,322 @@
...
+            -1,
+            31,
+            1,
+            0,
+            0,
+            0,
+            0,
+            SessionPtr}, // vaeDecodeOnly=true, NThreads=-1,
+                         // wtype=31(SD_TYPE_COUNT), RngType=CUDA_RNG,
+                         // Schedule=DEFAULT, Other is false
```
Comment (dannypsnl) [inline test/plugins/wasmedge_stablediffusion/wasmedge_stablediffusion.cpp:172]: same

File: plugins/CMakeLists.txt [hunk 3]
```diff
@@ -66,6 +66,10 @@ if(WASMEDGE_PLUGIN_WASI_OCR)
   add_subdirectory(wasi_ocr)
 endif()
 
+if(WASMEDGE_PLUGIN_STABLEDIFFUSION)
+    add_subdirectory(wasmedge_stablediffusion)
+endif()
```
Comment (dannypsnl) [inline plugins/CMakeLists.txt:71]: Tests only run on Linux, maybe we need to exclude platforms here as well?

**Review by dannypsnl — CHANGES_REQUESTED** [review]

**Review by hydai — APPROVED** [review]
LGTM. @dannypsnl PTAL

**Review by dannypsnl — APPROVED** [review]

**Pull request merged**